sdk/cosmos/azure-cosmos-kafka-connect/src/main/java/com/azure/cosmos/kafka/connect/implementation/source/MetadataMonitorThread.java

```diff
@@ -70,6 +71,10 @@ public MetadataMonitorThread(
         this.metadataReader = metadataReader;
         this.cosmosClient = cosmosClient;
         this.containersQuerySpec = this.getContainersQuerySpec();
+        int numContainers = Objects.requireNonNull(this.getAllContainers().block()).size();
```

Comment:
to be consistent with sink connector, can we also have this check in the source connector startup method vs here?

Reply:
moved it to be consistent with the sink connector.